grails-app/services/com/unifina/service/ProductService.groovy

```diff
@@ -13,6 +14,28 @@ class ProductService {
 	PermissionService permissionService
 	SubscriptionService subscriptionService
 
+	void removeUsersProducts(String username) {
+		def all = Product.findAllByOwner(username)
+		all.toArray().each{ Product product ->
+			product.delete(flush: true)
+			// Delete users all subscriptions
+			def user = SecUser.where {
+				username == username
```

Comment:
The idea is not to delete the troll user's subscriptions but rather all the subscriptions to the troll user's Products.